nistats/first_level_model.py

```diff
@@ -0,0 +1,561 @@
+
+            # Build the experimental design for the glm
+            run_img = check_niimg(run_img, ensure_ndim=4)
+            if design_matrices is None:
+                n_scans = run_img.get_data().shape[3]
+                if confounds is not None:
+                    confounds_matrix = confounds[run_idx].values
+                    if confounds_matrix.shape[0] != n_scans:
+                        raise ValueError('Rows in confounds does not match'
+                                         'n_scans in run_img at index %d'
+                                         % (run_idx,))
+                    confounds_names = confounds[run_idx].columns
+                else:
+                    confounds_matrix = None
+                    confounds_names = None
+                start_time = self.slice_time_ref * self.t_r
+                end_time = (n_scans - self.slice_time_ref) * self.t_r
+                frame_times = np.linspace(start_time, end_time, n_scans)
+                design = make_design_matrix(frame_times, paradigms[run_idx],
+                                            self.hrf_model, self.drift_model,
+                                            self.period_cut, self.drift_order,
+                                            self.fir_delays, confounds_matrix,
+                                            confounds_names, self.min_onset)
+            else:
+                design = design_matrices[run_idx]
+            self.design_matrices_.append(design)
+
+            # Compute GLM
+            Y = self.masker_.transform(run_img)
+            if self.signal_scaling:
+                Y, _ = mean_scaling(Y, self.scaling_axis)
+            if self.memory is not None:
+                mem_glm = self.memory.cache(run_glm)
+            else:
+                mem_glm = run_glm
+            labels, results = mem_glm(Y, design,
+                                      noise_model=self.noise_model,
+                                      bins=100, n_jobs=self.n_jobs)
+            self.labels_.append(labels)
+            # We save memory if inspecting model details is not necessary
+            if self.minimize_memory:
+                for key in results:
+                    results[key] = SimpleRegressionResults(results[key])
+            self.results_.append(results)
+            del Y
+
+        # Report progress
+        if self.verbose > 0:
+            sys.stderr.write("\nComputation of %d runs done in %i seconds\n"
+                             % (n_runs, time.time() - t0))
+
+        return self
+
+    def get_design_matrices(self):
+        """Get design matrices computed during model fit
+
+        Returns
+        -------
+        design_matrices : list of pandas DataFrames,
+            Holds the design matrices computed for the corresponding run_imgs
+        """
+        if self.design_matrices_ is None:
+            raise ValueError('The model has not been fit yet')
+        else:
+            return self.design_matrices_
```

Comment:
I think we should remove such getters that do exactly nothing.
writing model.design_matrices_ is not more complicated than model.get_design_matrices()